test/e2e/deployment/deployment.go

```diff
@@ -0,0 +1,50 @@
+package deployment
+
+import (
+	"context"
+	"time"
+
+	"github.com/onsi/gomega"
+	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	deputils "k8s.io/kubectl/pkg/util/deployment"
+)
+
+type DeploymentCommands struct{ deployment appsv1.Deployment }
+
+func (dc DeploymentCommands) ReadyConditionEventually() gomega.AsyncAssertion {
+	return gomega.Eventually(func() bool {
+		cond := deputils.GetDeploymentCondition(dc.deployment.Status, appsv1.DeploymentAvailable)
+		if cond == nil {
+			return false
+		}
+		return cond.Status == corev1.ConditionTrue
+	}, 120*time.Second, 1*time.Second)
+}
+
```

Comment:
This is not longer needed since we have the matcher, let's do it like the DamonSet